### PR TITLE
Gc

### PR DESCRIPTION
**Pull request by abargnesi**
addresses garbage collection of Redlander objects; refs #3

### CODE REVIEW

File: lib/redlander/node.rb [hunk 1]
```diff
@@ -132,5 +131,15 @@ def wrap(n)
...
       end
     end
+
+    # @api private
+    def self.finalize(rdf_node_ptr)
+      proc {
+        valid_ptr = Redlander.librdf_node_is_resource(rdf_node_ptr) or
+          Redlander.librdf_node_is_literal(rdf_node_ptr) or
+          Redlander.librdf_node_is_blank(rdf_node_ptr)
+        Redland.librdf_free_node(rdf_node_ptr) if valid_ptr
```
Comment (cordawyn) [inline lib/redlander/node.rb:141]: Why do you need this check? Checking for `valid_ptr.null?` (which is performed above in `initialize` method) should be sufficient to ensure validity of the "pointer", I think.

**Comment by cordawyn** [comment]
@abargnesi I've merged your pull request with some cosmetic changes. Please take a look at commit 7b8b9502de859bf121434ed704755800a7591647 and let me know if you have questions or objections. Thank you!

I renamed your `finalize` methods to `finalize_*` ones because `Model#finalize` that you suggested should not always call both `librdf_free_storage` and `librdf_free_model`, so I made two methods out of it, and then renamed the rest for conformity (yes, I know these method names are not very good - feel free to suggest something else).

File: lib/redlander/node.rb (hunk 1 above)
Comment (abargnesi) [inline lib/redlander/node.rb:141]: An `rdf_node_ptr.null?` expression always printed `false` even when I encountered invalid pointer stacktraces.  See [this gist](https://gist.github.com/abargnesi/11289193) for a test that shows invalid pointer issue.

**Comment by abargnesi** [comment]
The changes in 7b8b950 look good to me.  The method names seem appropriate.

**Pull request merged**